src/sstruct_mv/HYPRE_sstruct_vector.c

```diff
@@ -204,12 +204,16 @@ HYPRE_SStructVectorInitialize( HYPRE_SStructVector vector )
       ilower = hypre_SStructGridStartRank(grid);
       iupper = ilower + hypre_SStructGridLocalSize(grid) - 1;
    }
-
-   if (vector_type == HYPRE_SSTRUCT || vector_type == HYPRE_STRUCT)
+   else if (vector_type == HYPRE_SSTRUCT || vector_type == HYPRE_STRUCT)
    {
       ilower = hypre_SStructGridGhstartRank(grid);
       iupper = ilower + hypre_SStructGridGhlocalSize(grid) - 1;
    }
+   else
```

Comment:
Will you pull these `struct/sstruct` changes into `recmat`?

Reply:
Yes, I will! Hopefully, it won't be too much work :)